core/object/worker_thread_pool.cpp

```diff
@@ -496,6 +503,10 @@ bool WorkerThreadPool::is_group_task_completed(GroupID p_group) const {
 }
 
 void WorkerThreadPool::wait_for_group_task_completion(GroupID p_group) {
+#ifndef THREADS_ENABLED
+	return;
+#endif
```

Comment:
I think this should be an if/else/endif block instead, this would complain on some compilers due to unreachable code